File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3SiriDownstreamSubscriptionResponse.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3SiriDownstreamSubscriptionResponse {
    /// <summary>
    /// The Data Horizon of Chronos
    /// </summary>
    /// <value>The Data Horizon of Chronos</value>
    [DataMember(Name="valid_until", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "valid_until")]
    public DateTime? ValidUntil { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3SiriDownstreamSubscriptionResponse {\n");
      sb.Append("  ValidUntil: ").Append(ValidUntil).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: dart/lib/model/v3_siri_downstream_subscription_response.dart
part of swagger.api;

class V3SiriDownstreamSubscriptionResponse {
  /* The Data Horizon of Chronos */
  DateTime validUntil = null;

  V3SiriDownstreamSubscriptionResponse();

  @override
  String toString() {
    return 'V3SiriDownstreamSubscriptionResponse[validUntil=$validUntil, ]';
  }

  V3SiriDownstreamSubscriptionResponse.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    validUntil = json['valid_until'] == null ? null : DateTime.parse(json['valid_until']);
  }

  Map<String, dynamic> toJson() {
    return {
      'valid_until': validUntil == null ? '' : validUntil.toUtc().toIso8601String()
     };
  }

  static List<V3SiriDownstreamSubscriptionResponse> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3SiriDownstreamSubscriptionResponse>() : json.map((value) => new V3SiriDownstreamSubscriptionResponse.fromJson(value)).toList();
  }

  static Map<String, V3SiriDownstreamSubscriptionResponse> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3SiriDownstreamSubscriptionResponse>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3SiriDownstreamSubscriptionResponse.fromJson(value));
    }
    return map;
  }
}
